packages/components/src/color-picker/test/index.tsx

```diff
@@ -81,8 +84,11 @@ describe( 'ColorPicker', () => {
 			);
 
 			const saturation = getSaturation( container );
+
+			expect( saturation ).toBeInTheDocument();
+
 			moveReactColorfulSlider(
-				saturation,
+				saturation!,
```

Comment:
The element here could be `null` if `querySelector` came up empty, so we should check for that.

To the best of my knowledge (though I'm happy if I'm wrong) TypeScript can't narrow based on test failures, so we need to narrow explicitly rule out `null` values now that they'll have been handled by the `toBeInTheDocument()` check.

This pattern is used twice more in subsequent tests, if a change is needed, I'll apply it to all three cases.

Reply:
I wonder if adding an explicit check + throwing error may work (ie. the test would need to fail if the element is `null`)

```diff
diff --git a/packages/components/src/color-picker/test/index.tsx b/packages/components/src/color-picker/test/index.tsx
index 2820710268..80c287081e 100644
--- a/packages/components/src/color-picker/test/index.tsx
+++ b/packages/components/src/color-picker/test/index.tsx
@@ -85,10 +85,14 @@ describe( 'ColorPicker', () => {
 
 			const saturation = getSaturation( container );
 
+			if ( saturation === null ) {
+				throw new Error( 'The saturation slider could not be found' );
+			}
+
 			expect( saturation ).toBeInTheDocument();
 
 			moveReactColorfulSlider(
-				saturation!,
+				saturation,
 				{ pageX: 0, pageY: 0 },
 				{ pageX: 10, pageY: 10 }
 			);

```

Reply:
Smart, seems to work great. Thanks! f33002c694